utils.go

```diff
@@ -16,18 +17,28 @@ const (
 	TagsHTTPHeaderPrefix = "Open-Tracing-Trace-Tags-"
 )
 
-// PropagateSpanInHeader encodes Span `ctx` to `h` as a series of
-// HTTP headers. Values are URL-escaped.
-func PropagateSpanInHeader(
+// InjectSpanInHeader encodes Span `sp` in `h` as a series of HTTP headers.
+// Values are URL-escaped.
+func InjectSpanInHeader(
 	sp Span,
 	h http.Header,
-	propagator SpanPropagator,
-) {
-	contextIDMap, tagsMap := propagator.PropagateSpanAsText(sp)
-	for headerSuffix, val := range contextIDMap {
+) error {
+	// First, look for a PROPAGATION_FORMAT_GO_HTTP_HEADER injector (our preference).
+	injector := sp.PropagationInjectorForFormat(PROPAGATION_FORMAT_GO_HTTP_HEADER)
```

Comment:
@bensigelman 
> As for the "carriers": the motivating question is how the OpenTracing user who is dealing with process-boundary concerns (i.e., the person writing code like the lines we're commenting about here) is expected to pass in the http.Header or map[string]string (or whatever) that they're trying to inject into. I know that you wanted to do this – in Java, that is – by having the Injector rely on introspection

There was no introspection required in my code example. And neither any need for variadic parameters.

What I'm looking for is the delegation model of

    HttpHeaderSpanInjector --> SplitTextInjector

where the end-user only needs to call `span.Injector(opentracing.HttpHeader).Inject(span, httpReq.Header)`